dist/src/test/java/io/camunda/webapps/controllers/IdentityClientConfigControllerTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.webapps.controllers;
+
+import static com.google.common.net.HttpHeaders.CONTENT_SECURITY_POLICY;
+import static org.assertj.core.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.identity.webapp.controllers.IdentityClientConfigController;
+import io.camunda.security.configuration.AuthenticationConfiguration;
+import io.camunda.security.configuration.OidcAuthenticationConfiguration;
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.security.entity.AuthenticationMethod;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.web.servlet.DispatcherServletPath;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest(controllers = IdentityClientConfigController.class)
+@Import(IdentityClientConfigControllerTest.TestConfig.class)
+class IdentityClientConfigControllerTest {
+
+  @Autowired private MockMvc mockMvc;
+
+  @Test
+  void shouldReturnClientConfigWithOidcTrue() throws Exception {
+    final String response =
+        mockMvc
+            .perform(get("/identity/config.js"))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType("text/javascript;charset=UTF-8"))
+            .andExpect(header().doesNotExist(CONTENT_SECURITY_POLICY))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+    final ObjectMapper objectMapper = new ObjectMapper();
+    final var configResponse =
+        objectMapper.readValue(
+            response.substring(response.indexOf('{'), response.lastIndexOf('}') + 1), Map.class);
```

Comment:
Can you please extract the indexes first and assert that they are `!= -1` before you use them? That should make for a better readable failure message whenever we break the test in the future.

Reply:
Done, if I understood you correctly